test/system/src/main/java/io/pravega/test/system/framework/services/docker/PravegaSegmentStoreDockerService.java

```diff
@@ -0,0 +1,143 @@
+import com.spotify.docker.client.messages.swarm.ServiceMode;
+import com.spotify.docker.client.messages.swarm.ServiceSpec;
+import com.spotify.docker.client.messages.swarm.TaskSpec;
+import lombok.extern.slf4j.Slf4j;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.TimeoutException;
+import static org.hamcrest.core.Is.is;
+import static org.hamcrest.core.IsNull.notNullValue;
+import static org.junit.Assert.assertThat;
+
+@Slf4j
+public class PravegaSegmentStoreDockerService extends DockerBasedService {
+
+    private static final int SEGMENTSTORE_PORT = 12345;
+    private static final String SEGMENTSTORE_EXTRA_ENV = System.getProperty("segmentStoreExtraEnv");
+    private static final String ENV_SEPARATOR = ";;";
+    private static final java.lang.String KEY_VALUE_SEPARATOR = "::";
+    private int instances = 1;
+    private double cpu = 0.1;
+    private double mem = 1000.0;
+
+    public PravegaSegmentStoreDockerService(final String serviceName) {
+        super(serviceName);
+    }
+
+    @Override
+    public void stop() {
+        try {
+            dockerClient.removeService(getID());
+        } catch (DockerException | InterruptedException e) {
+            log.error("unable to remove service {}", e);
+        }
+    }
+
+    @Override
+    public void clean() {
+    }
+
+    @Override
+    public void start(final boolean wait) {
+        try {
+            ServiceCreateResponse serviceCreateResponse = dockerClient.createService(setServiceSpec());
+            if (wait) {
+                waitUntilServiceRunning().get(5, TimeUnit.MINUTES);
+            }
+            assertThat(serviceCreateResponse.id(), is(notNullValue()));
+        } catch (InterruptedException | DockerException | TimeoutException | ExecutionException e) {
+            log.error("unable to create service {}", e);
+        }
+    }
+
+    private ServiceSpec setServiceSpec() {
+        Mount mount = Mount.builder().type("volume").source("logs-volume").target("/tmp/logs").build();
+        //set env
+        String zk = "zookeeper:" + ZKSERVICE_ZKPORT;
+        String con = "controller:" + CONTROLLER_PORT;
+        //System properties to configure SS service.
+        String hostSystemProperties =
+                setSystemProperty("autoScale.muteInSeconds", "120") +
+                setSystemProperty("autoScale.cooldownInSeconds", "120") +
+                setSystemProperty("autoScale.cacheExpiryInSeconds", "120") +
+                setSystemProperty("autoScale.cacheCleanUpInSeconds", "120") +
+                setSystemProperty("log.level", "DEBUG") +
+                setSystemProperty("curator-default-session-timeout", String.valueOf(30 * 1000))+
+                        setSystemProperty("hdfs.replaceDataNodesOnFailure", "false");
+
+        String env1 = "PRAVEGA_SEGMENTSTORE_OPTS=" + hostSystemProperties;
```

Comment:
These look to only be used once. Perhaps they should be inlined.

Reply:
done.